tracer/src/Datadog.Trace/DuckTyping/DuckType.cs

```diff
@@ -620,15 +620,39 @@ private static void CreateProperties(TypeBuilder? proxyTypeBuilder, Type proxyDe
                 {
                     case DuckKind.Property:
                         PropertyInfo? targetProperty = null;
-                        try
+                        if (duckAttribute.Name.IndexOf(',') == -1)
                         {
-                            targetProperty = targetType.GetProperty(duckAttribute.Name, duckAttribute.BindingFlags);
+                            try
+                            {
+                                targetProperty = targetType.GetProperty(duckAttribute.Name, duckAttribute.BindingFlags);
+                            }
+                            catch
+                            {
+                                // This will run only when multiple indexers are defined in a class, that way we can end up with multiple properties with the same name.
+                                // In this case we make sure we select the indexer we want
+                                targetProperty = targetType.GetProperty(duckAttribute.Name, proxyProperty.PropertyType, proxyProperty.GetIndexParameters().Select(i => i.ParameterType).ToArray());
+                            }
                         }
-                        catch
+                        else
                         {
-                            // This will run only when multiple indexers are defined in a class, that way we can end up with multiple properties with the same name.
-                            // In this case we make sure we select the indexer we want
-                            targetProperty = targetType.GetProperty(duckAttribute.Name, proxyProperty.PropertyType, proxyProperty.GetIndexParameters().Select(i => i.ParameterType).ToArray());
+                            foreach (var name in duckAttribute.Name.Split(','))
+                            {
+                                try
+                                {
+                                    targetProperty = targetType.GetProperty(name, duckAttribute.BindingFlags);
+                                }
+                                catch
+                                {
+                                    // This will run only when multiple indexers are defined in a class, that way we can end up with multiple properties with the same name.
+                                    // In this case we make sure we select the indexer we want
+                                    targetProperty = targetType.GetProperty(name, proxyProperty.PropertyType, proxyProperty.GetIndexParameters().Select(i => i.ParameterType).ToArray());
+                                }
```

Comment:
Is it worth moving the duplicated code to a local function do you think?

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/3616/commits/faa23646d03e2a672a2cd3f22e3b25c0e2838ce3